src/driver/test/workload_parsers_test.cpp

```diff
@@ -57,5 +102,59 @@ TEST_CASE("WorkloadParser can run generate smoke test configurations") {
     REQUIRE(YAML::Dump(parsedConfig) == expected);
 }
 
+TEST_CASE("WorkloadParser can preprocess keywords") {
+    const auto input = (R"(
+ActorTemplates:
+- TemplateName: TestTemplate
+  Config:
+    Name: {^Parameter: {Name: "Name", Default: "IncorrectDefault"}}
+    SomeKey: SomeValue
+    Phases:
+      OnlyIn:
+        Active: [{^Parameter: {Name: "Phase", Default: 1}}]
+        Max: 3
```

Comment:
I don't get what `Max` is here - is that documented more explicitly somewhere? If it wants to be active in more than 1 Phase, can it be:

```yaml
ActiveInPhases: [{^Parameter: {Name: ActiveInPhases, Default: 3}]
# Or:
ActiveInPhases: All
# Or:
OnlyActiveInPhaseRange:
  From: {^Parameter: {Name: ActiveFromPhase, Default: 7}
  To: {^Parameter: {Name: ActiveToPhase, Default: 10}
# Or:
```

And then enforce that `ActiveInPhases` is not empty xor `ActiveInPhases==All` xor `ActiveInPhaseRange` is non-empty.

Would hopefully be a simple change to extract the logic:

```c++
std::pair<int, int> extractPhaseRange(YAML::Node phasesBlock); // => min/max inclusive
```

Reply:
`Max` lets the evaluator know the total number of phases that are going to run, so that inactive phases can be listed as `Nop`. I'd originally hoped that we could just not list phases in genny and they'll be assumed a Nop, but that causes a throw [here](https://github.com/mongodb/genny/blob/8f22265a48b6c9fd56ffb35c428db6e13a7a5cff/src/gennylib/include/gennylib/PhaseLoop.hpp#L491).

Another option would be to do a second preprocessing pass after all the templates and actors have been evaluated to determine how many phases will run, but I feel like it's hairy to have the result of evaluation rely on "non-local" information.

Reply:
Oh, each Actor must know the total number of Phases across all other Actors. Gotcha. Hm. I'm not super keen on modifying `Context` just yet.

Maybe rename to `NopInPhasesUpTo`?

```yaml
Phases:
  OnlyActiveInPhases: [1]
  NopInPhasesUpTo: 2
  PhaseConfig: {foo: bar}
```

Equivalent to:

```yaml
Phases:
- {Phase: 0, Nop: true}
- {Phase: 1, foo: bar}
- {Phase: 2, Nop: true}
```
 Naming is hard.

Reply:
Good idea. Done!